jl4-lsp/src/LSP/L4/Viz/Ladder.hs

```diff
@@ -133,12 +163,22 @@ translateExpr False = go
           V.Or uid <$> traverse go (scanOr e)
         Where _ e' _ds -> go e' -- TODO: lossy
 
+        -- TODO: Should handle BoolLits differently too
         -- 'var'
         App _ resolved [] ->
           leafFromVizName (mkVizNameWith prettyLayout resolved)
-        -- TODO: Will be replacing this temporary version with a variant for App on the frontend
-        App _ (getOriginal -> MkName _ fnName) args ->
-          leaf "" $ Text.unwords (rawNameToText fnName : (prettyLayout <$> args))
+
+        fnApp@(App _ fnResolved@(getOriginal -> fnName) args) -> do
```

Comment:
I think this is goo complicated. there's not much point in using a view pattern that doesn't do any matching. Also if you see the comment above, you should probably simplify the at pattern.

Reply:
Good point, thanks!